pkg/config/autodiscovery/autodiscovery.go

```diff
@@ -77,8 +77,8 @@ func DiscoverComponentsFromEnv() ([]config.ConfigurationProviders, []config.List
 	detectedListeners = append(detectedListeners, config.Listeners{Name: "environment"})
 	detectedListeners = append(detectedListeners, config.Listeners{Name: "static config"})
 
-	// Automatic handling of AD providers/listeners should only run in Core agent.
-	if flavor.GetFlavor() != flavor.DefaultAgent {
+	// Automatic handling of AD providers/listeners should only run in the core or process agent.
+	if flavor.GetFlavor() != flavor.DefaultAgent || flavor.GetFlavor() != flavor.ProcessAgent {
```

Comment:
```suggestion
	if flavor.GetFlavor() != flavor.DefaultAgent && flavor.GetFlavor() != flavor.ProcessAgent {
```

If I'm understanding the comment correctly, it should return early if we are not the core agent and we are also not the process agent, right?

Reply:
Ah, yes, it appears I forgot De Morgan's law 😅